samples/CppArExample/CppArExample.cpp

```diff
@@ -310,6 +317,45 @@ void CppArExample::createTabletopTestScene()
   changeScene();
 }
 
+// If m_screenToLocationMode is true, create and place a 3D sphere in the scene. Otherwise set the
+// initial transformation based on the screen position.
+void CppArExample::onTouched(QTouchEvent& event)
+{
+  Q_CHECK_PTR(m_arcGISArView);
+
+  const QPoint screenPoint = event.touchPoints().first().lastScreenPos().toPoint();
```

Comment:
This looks dangerous since it assumes `first()` returns a valid object, but I think you're safe since `onTouched` would only be invoked for an actual touch event.

Do you think there's any need to validate this for safety? I think it's ok like this.

Reply:
Doc says it's possible to "cancel" a touch point. I don't know how to do that on mobile device. But you're right, it's safer to check if the list is not empty.